src/freenet/node/AnnounceSender.java

```
@@ -484,6 +484,15 @@ private void timedOut(PeerNode next) {
 	}
 
 	private void rnf(PeerNode next) {
+		synchronized(this) {
+			while(waitingForTransfers > 0) {
+				try {
+					wait();
+				} catch (InterruptedException e) {
+					// Ignore.
```

I’m not a big fan of ignoring exceptions… is this acceptable here?